awx/main/tasks.py

```diff
@@ -3028,12 +3055,17 @@ def run(self):
         # We establish a connection to the Receptor socket
         receptor_ctl = get_receptor_ctl()
 
+        res = None
         try:
-            return self._run_internal(receptor_ctl)
+            res = self._run_internal(receptor_ctl)
+            return res
         finally:
             # Make sure to always release the work unit if we established it
             if self.unit_id is not None and settings.RECEPTOR_RELEASE_WORK:
                 receptor_ctl.simple_command(f"work release {self.unit_id}")
+            # If an error occured without the job itself failing, it could be a broken instance
+            if self.work_type == 'ansible-runner' and res is None or getattr(res, 'rc', None) is None:
+                execution_node_health_check(self.task.instance.execution_node)
```

Comment:
@shanemcd I'd like for you to think about this.

For healthy execution nodes _we don't regularly re-run health checks_. But what if one of them was working, but is no longer working? We shouldn't keep sending jobs there. So we have 2 options, we can run checks on a regular basis, or we can re-run health checks when a job on that instance gets a suspicious error.

We don't want to do this when the ansible-playbook sub-process errors (have an `rc`), because that's on the execution layer, not the instance configuration layer (as far as we know).